internal/config/cgmgr/stats_linux.go

```diff
@@ -20,22 +20,49 @@ import (
 	types "k8s.io/cri-api/pkg/apis/runtime/v1"
 )
 
-// CgroupStats is a structure used to augment libctr.Stats object, because
-// it does not have all of the information required for all of the stats we're
-// interested in.
+// This is a universal stats object to be used across different runtime implementations.
```

Comment:
I find myself worried about the translation here. there is already high cpu usage on stats collection from golang object GC (each tiny object is created and destroyed quite quickly). I think I am okay with this for now but it's possible we may want to find a different way. For instance, having the statsserver run linux only for now

Reply:
If you'd rather do that as a follow up (or not deal with that and have someone else do it) @Mo-Fatah let me know

Reply:
I personally wanted to go the way Sohan did before [here](https://github.com/cri-o/cri-o/pull/7256/files#diff-d9db4f1729b93e9ec3b256488380d0ee3b8b06756cfc30d406b9de6976c30779R22-R25) to just use the , but due to the newly-added FreeBSD build workflow, apparently any inclusion of libcontainer in runtime_vm.go (through importing libcontainer or adding a type that uses a libcontainer type internally) will lead to FreeBSD build failure (at least this what I noticed from multiple CI failures).

 I would prefer to solve this in a separate PR if it's Ok.

Reply:
>I would prefer to solve this in a separate PR if it's Ok.
That sounds good!